Model/Payment_Method.php

```diff
@@ -166,4 +166,30 @@ public function getConfigFields()
             ],
         ];
     }
+
+    /**
+     * Returns array of setting names which should not be part of output debug infos
+     *
+     * @return array
+     */
+    public function getProtectedFieldNames()
+    {
+        return ['httpUser', 'httpPassword', 'secret'];
+    }
```

Comment:
Should we maybe use a whitelist instead to reduce the risk that an unwanted field slips through  in the future?

Reply:
I have changed the logic to use getPublicFieldNames.